### ISSUE TITLE
Device states stop updating few minutes after tuya start. “Caught exception in on_message”

### ISSUE THREAD

**Issue by ravac1** [tuya/tuya-home-assistant#230]
**Describe the bug**
Device states don’t update anymore after working for some time (minutes). 

**Expected behavior**
States should be the same as in SmartLife app. 

**Home assistant version**
2021.7.4

**Additional context**
The logs when the issue appears:

2021-07-25 23:49:28 DEBUG (Thread-5) [custom_components.tuya_v2] _update--><custom_components.tuya_v2._init_tuya_sdk.<locals>.DeviceListener object at 0x6eec6310>;->>{'pir': 'none', 'battery_percentage': 100, 'temper_alarm': False, 'bright_value': 0, 'pir_sensitivity': 'middle', 'pir_time': '60s'}
2021-07-25 23:49:28 DEBUG (Thread-5) [tuya iot] _on_log: Caught exception in on_message: 'NoneType' object has no attribute 'add_job'

And trace from a new occurence:

Logger: root
Source: helpers/entity.py:559 
First occurred: 0:15:21 (4 occurrences) 
Last logged: 6:23:01

Uncaught thread exception
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/threading.py", line 954, in _bootstrap_inner
    self.run()
  File "/usr/local/lib/python3.9/threading.py", line 892, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/local/lib/python3.9/site-packages/paho/mqtt/client.py", line 3452, in _thread_main
    self.loop_forever(retry_first_connection=True)
  File "/usr/local/lib/python3.9/site-packages/paho/mqtt/client.py", line 1779, in loop_forever
    rc = self.loop(timeout, max_packets)
  File "/usr/local/lib/python3.9/site-packages/paho/mqtt/client.py", line 1181, in loop
    rc = self.loop_read(max_packets)
  File "/usr/local/lib/python3.9/site-packages/paho/mqtt/client.py", line 1572, in loop_read
    rc = self._packet_read()
  File "/usr/local/lib/python3.9/site-packages/paho/mqtt/client.py", line 2310, in _packet_read
    rc = self._packet_handle()
  File "/usr/local/lib/python3.9/site-packages/paho/mqtt/client.py", line 2936, in _packet_handle
    return self._handle_publish()
  File "/usr/local/lib/python3.9/site-packages/paho/mqtt/client.py", line 3216, in _handle_publish
    self._handle_on_message(message)
  File "/usr/local/lib/python3.9/site-packages/paho/mqtt/client.py", line 3444, in _handle_on_message
    self.on_message(self, self._userdata, message)
  File "/usr/local/lib/python3.9/site-packages/tuya_iot/openmq.py", line 157, in _on_message
    listener(msgDict)
  File "/usr/local/lib/python3.9/site-packages/tuya_iot/device.py", line 175, in _on_message
    self._on_device_report(data["devId"], data["status"])
  File "/usr/local/lib/python3.9/site-packages/tuya_iot/device.py", line 196, in _on_device_report
    self.__update_device(device)
  File "/usr/local/lib/python3.9/site-packages/tuya_iot/device.py", line 183, in __update_device
    listener.update_device(device)
  File "/config/custom_components/tuya_v2/__init__.py", line 163, in update_device
    ha_device.schedule_update_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 559, in schedule_update_ha_state
    self.hass.add_job(self.async_update_ha_state(force_refresh))  # type: ignore
AttributeError: 'NoneType' object has no attribute 'add_job'

**Comment by sodich** [comment]
the same for me

**Comment by ravac1** [comment]
as workaround, modify config/custom_components/tuya_v2/init.py, line 163

```
try:
    ha_device.schedule_update_ha_state()
except:
    _LOGGER.error(f"print what you want")
```

**Comment by sodich** [comment]
Didn't work as well, It's getting the correct temperature after the restart then sticks on it forever or until the next restart.

**Comment by tsutsuku** [comment]
@ravac1 Try branch master, we got a ticket today and fix it. Maybe it's the same reason why the error happens.

**Comment by ravac1** [comment]
> @ravac1 Try branch master, we got a ticket today and fix it. Maybe it's the same reason why the error happens.

Thanks. I’ll try it in the next days

**Comment by ravac1** [comment]
I use master branch now and so far it works well.

**Comment by ravac1** [comment]
This seems to be fixed on 1.5.0
